src/VisualStudio/Core/Impl/CodeModel/FileCodeModel.cs

```diff
@@ -541,7 +542,7 @@ internal EnvDTE.CodeElement CodeElementFromPosition(int position, EnvDTE.vsCMEle
                     // If both tokens are touching, we prefer identifiers and keywords to
                     // separators. Note that the language doesn't allow both tokens to be a
                     // keyword or identifier.
-                    if (SyntaxFactsService.IsKeyword(rightToken) ||
+                    if (SyntaxFactsService.IsReservedOrContextualKeyword(rightToken) ||
                         SyntaxFactsService.IsIdentifier(rightToken))
```

Comment:
note: we probably just want to use .IsWord here, but i didn't want to change semantics in this PR.